sqlmesh/core/engine_adapter/shared.py

```diff
@@ -47,3 +55,74 @@ class DataObject(PydanticModel):
     schema_name: str = Field(alias="schema")
     name: str
     type: DataObjectType
+
+
+def set_catalog(
+    *,
+    override: t.Optional[CatalogSupport] = None,
+) -> t.Callable:
+    def decorator(func: t.Callable) -> t.Callable:
+        @functools.wraps(func)
+        def wrapper(*args: t.Any, **kwargs: t.Any) -> t.Any:
+            def _get_args_pos_and_kwarg_name(
```

Comment:
why is this nested? does it need  to be? makes it hard to read this

Reply:
Moved out